dante/osomerank/elicited_response.py

```diff
@@ -48,38 +49,48 @@ def load_er_models():
-                )
-                model = AutoModelForSequenceClassification.from_pretrained(
-                    pretrained_model_name_or_path=model_path,
-                    num_labels=1,
-                    ignore_mismatched_sizes=True,
-                )
-                pipeline = TextClassificationPipeline(
-                    model=model,
-                    tokenizer=tokenizer,
-                    max_length=512,
-                    truncation=True,
-                    batch_size=8,
-                    # top_k=None,
-                    # device="cuda",  # uncomment if GPU is available
-                )
-                MODEL_PIPELINES[f"{model_name}_{platform}"] = pipeline
-                logger.info(f"Loaded {model_name}_{platform} model.")
-            except Exception as e:
-                logger.error(f"Failed to load {model_name}_{platform} model: {e}")
+
+    try:
+        prefix = config.get("ELICITED_RESPONSE", "elicited_response_dir")
+        er_model_dir = os.path.join(cache_path, prefix)
+        if not os.path.exists(er_model_dir) or not os.listdir(er_model_dir):
+            logger.warning("No cached model found! Retrieving from S3.")
+            fetchfroms3(prefix, cache_path)
+        # load MODEL_PIPELINES
+        for model_name in model_names:
+            for platform in platforms:
+                try:
+                    mod_prefix = config.get(
+                        "ELICITED_RESPONSE", f"{model_name}_{platform}"
+                    )
+                    model_path = os.path.join(cache_path, mod_prefix)
+                    tokenizer = AutoTokenizer.from_pretrained(
+                        pretrained_model_name_or_path=model_path
+                    )
+                    model = AutoModelForSequenceClassification.from_pretrained(
+                        pretrained_model_name_or_path=model_path,
+                        num_labels=1,
+                        ignore_mismatched_sizes=True,
+                    )
+                    pipeline = TextClassificationPipeline(
+                        model=model,
+                        tokenizer=tokenizer,
+                        max_length=512,
+                        truncation=True,
+                        batch_size=8,
+                        # top_k=None,
+                        # device="cuda",  # uncomment if GPU is available
+                    )
+                    MODEL_PIPELINES[f"{model_name}_{platform}"] = pipeline
+                    logger.info(f"Loaded {model_name}_{platform} model.")
+                except (configparser.NoSectionError, OSError) as e:
+                    logger.error(f"Failed to load {model_name}_{platform} model: {e}")
+                except Exception as e:
+                    logger.error(f"Failed to load {model_name}_{platform} model: {e}")
+
+    except Exception as e:
+        logger.error(f"Failed to load all elicited_response models: {e}")
+        raise
```

Comment:
Right now this is going to log and raise any error, if it happens between L53 and L62. Once you are in the inner try / except, errors that occur there will only be logged but not raised. Is this what you want?